pkg/kv/kvserver/replica.go

```diff
@@ -1741,14 +1738,11 @@ func (r *Replica) isNewerThanSplit(split *roachpb.SplitTrigger) bool {
 
 func (r *Replica) isNewerThanSplitRLocked(split *roachpb.SplitTrigger) bool {
 	rightDesc, _ := split.RightDesc.GetReplicaDescriptor(r.StoreID())
-	// If we have written a tombstone for this range then we know that the RHS
-	// must have already been removed at the split replica ID.
-	return r.mu.tombstoneMinReplicaID != 0 ||
-		// If the first raft message we received for the RHS range was for a replica
-		// ID which is above the replica ID of the split then we would not have
-		// written a tombstone but we will have a replica ID that will exceed the
-		// split replica ID.
-		r.replicaID > rightDesc.ReplicaID
+	// If the first raft message we received for the RHS range was for a replica
```

Comment:
The way I square this is that the `RaftReplicaID` and `RangeTombstone` are sort of two sides of the same coin. If you have a `RaftReplicaID` = `N`, it obviates the need for a `RangeTombstone` for `NextReplicaID=N`. So maybe this comment could be rephrased in those terms: we don't need to check a tombstone here since we are in a path where a replica exists. If a replica exists, tombstones don't matter because the replicaID always implies them.

Reply:
Nice framing. I think that this place now doesn't need a comment at all, it's self-explanatory. Maybe we should delete it, or move it (in this or your interpretation) somewhere closer to the caller side.

Reply:
Leaving as is for now, but maybe will make another pass at this comment, and the TODO above which seems done.